Weird Scala 2.13 Error: illegal trait super target found for method

## reproduction steps

using Scala 2.13.7,

```scala
class Outcome
trait TestSuite { thisTestSuite =>
  protected trait NoArgTest extends (() => Outcome)
  protected def withFixture(test: NoArgTest): Outcome = {
    test()
  }
}
trait TestSuiteMixin { this: TestSuite =>
  protected def withFixture(test: NoArgTest): Outcome
}
trait TimeLimitedTests extends TestSuiteMixin { this: TestSuite =>
  abstract override def withFixture(test: NoArgTest): Outcome = super.withFixture(test)
}
trait AnyFunSuiteLike extends TestSuite { thisSuite => }
abstract class Test[C] extends AnyFunSuiteLike with TimeLimitedTests
```

## problem

It got the following compiler error when compiles with Scala 2.13: 

```
cheeseng@cheeseng-RAVEN:~/git/213-bug$ scalac Test.scala
Test.scala:15: error: illegal trait super target found for method withFixture required by trait TimeLimitedTests;
 found   : protected def withFixture: ((test: _1.NoArgTest): Outcome) forSome { val _1: [C]Test[C] } in trait TestSuite;
 expected: protected def withFixture: ((test: _1.NoArgTest): Outcome) forSome { val _1: [C]Test[C] } in trait TestSuiteMixin
abstract class Test[C] extends AnyFunSuiteLike with TimeLimitedTests
```

The same code compiles fine with Scala 3 and 2.12.

This is originally reported as ScalaTest issue here: https://github.com/scalatest/scalatest/issues/2088

--- a/src/compiler/scala/tools/nsc/transform/Mixin.scala
+++ b/src/compiler/scala/tools/nsc/transform/Mixin.scala
@@ -103,10 +103,9 @@ abstract class Mixin extends Transform with ast.TreeDSL with AccessorSynthesis {
    *  @param acc        The symbol statically referred to by the superaccessor in the trait
    *  @param mixinClass The mixin class that produced the superaccessor
    */
-  private def rebindSuper(base: Symbol, acc: Symbol, mixinClass: Symbol): Symbol = {
-    val site = base.thisType
-
+  private def rebindSuper(base: Symbol, acc: Symbol, mixinClass: Symbol): Symbol =
     exitingSpecialize {
+      val site = base.thisType
       // the specialized version T$sp of a trait T will have a super accessor that has the same alias
       // as the super accessor in trait T; we must rebind super
       // from the vantage point of the original trait T, not the specialized T$sp
@@ -138,7 +137,6 @@ abstract class Mixin extends Transform with ast.TreeDSL with AccessorSynthesis {
 
       sym
     }
-  }
 
 // --------- type transformation -----------------------------------------------
 

--- a/test/files/pos/t12520.scala
+++ b/test/files/pos/t12520.scala
@@ -1,0 +1,16 @@
+class Outcome
+trait TestSuite {
+  protected trait NoArgTest extends (() => Outcome)
+  protected def withFixture(test: NoArgTest): Outcome = test()
+}
+
+trait TestSuiteMixin { this: TestSuite =>
+  protected def withFixture(test: NoArgTest): Outcome
+}
+
+trait TimeLimitedTests extends TestSuiteMixin { this: TestSuite =>
+  abstract override def withFixture(test: NoArgTest): Outcome = super.withFixture(test)
+}
+
+trait AnyFunSuiteLike extends TestSuite
+abstract class Test[C] extends AnyFunSuiteLike with TimeLimitedTests

